server.js

```diff
@@ -4288,13 +4288,14 @@ function(data, match, end, ask) {
   // Badge creation.
   try {
     var badgeData = getBadgeData(subject, data);
-    badgeData.colorscheme = undefined;
     if (data.label !== undefined) { badgeData.text[0] = '' + data.label; }
     badgeData.text[1] = status;
-    if (sixHex(color)) {
-      badgeData.colorB = '#' + color;
-    } else {
-      badgeData.colorscheme = color;
+    if ((data.colorB === undefined) && (data.colorA === undefined)) {
```

Comment:
If we had, say, `/badge/so-colourful-fa4391.svg?colorA=009900`, we'd expect to have a somewhat pink right-hand-side, and a somewhat dark green left-hand-side. Instead, this block is not executed, and the right-hand-side is green.

We only want to modify `colorB` (in the `if (sixHex(color)) {…}` branch) if the query parameter `colorB` is not defined, and that's it. In the `else` branch, we only want to set the `colorscheme` if neither `colorA` and `colorB` are defined as a query parameter.